### ISSUE TITLE
(Urgent)training/validation/evaluation label data input not consistent

### ISSUE THREAD

**Issue by Litchilitchy** [intel-analytics/analytics-zoo#1170]
Currently training label only support one-hot representation. (2-d input array), validation and evaluation label only support integer representation (1-d input array). 

e.g. 2 samples, with class label 0,2 respectively, training label input need to be `[ [1,0,0], [0,0,1] ]`, validation label input need to be `[0,2]`

**The most urgent one here is that we want to make then consistent (In Keras they are both one-hot vectors)**

_below is zoo current behavior and suggestion_

if training label use integer representation, it would run but gives very wrong process and result. (Suggest to raise an error here because it is hard to target this source issue when getting wrong result)

if validation or evaluation use one-hot representation, it would raise error `inconsistent tensor size`, saying e.g. 8 == 80, we inferred that 8 = batch_size 32 / kernel 4 and 80 is length=10 one-hot vector. (Also not very straight forward to target this issue because it maybe difficult for user to infer)

**Comment by Litchilitchy** [comment]
fixed by zhichao [here](https://github.com/intel-analytics/analytics-zoo/pull/1179)
